src/foam/net/ip/ClientIPAddressInfoFetch.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.net.ip',
+  name: 'ClientIPAddressInfoFetch',
+  documentation: `
+    Fetches IP address info using the method provided by the ipAddress 
+  `,
+
+  requires: [
+    'foam.net.ip.IPAddressInfo'
+  ],
+
+  properties: [
+    {
+      class: 'URL',
+      name: 'ipInfoProvider',
```

Comment:
@zahnno can you consider or investigate that the final real 'fetch' will fail early if one is running localhost with no network connection for example. Thanks.

Reply:
Otherwise, looks good.

Reply:
I could check if the user is connected to the internet but I'm assuming this consideration is deeper than that. Got any other info?

Reply:
perhaps the client nspec configuration can be empty for foam/src so it does nothing on localhost and CI, and we can configure it when not localhost.  and then, as a developer, you can still set it up locally to test.

Reply:
empty nspec - hmm, any calls to it would throw a no function error unless we abstract it to a class with an noop method.

Reply:
the url can be empty, and the fetch does nothing when undefined.